src/jit/morph.cpp

```diff
@@ -8246,14 +8246,22 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
 
     // Couldn't inline - remember that this BB contains method calls
 
-    // If this is a 'regular' call, mark the basic block as
-    // having a call (for computing full interruptibility).
+    // Mark the block as a GC safe point for the call if possible.
+    // In the event the call indicates the block isn't a GC safe point
+    // and the call is unmanaged with a GC transition suppression request
+    // then insert a GC poll.
     CLANG_FORMAT_COMMENT_ANCHOR;
 
     if (IsGcSafePoint(call))
     {
         compCurBB->bbFlags |= BBF_GC_SAFE_POINT;
     }
+    else if (call->IsUnmanaged() && call->IsSuppressGCTransition() && ((compCurBB->bbFlags & BBF_GC_SAFE_POINT) == 0))
```

Comment:
Based upon how "conservative" we (Jan & David) want this to be this should instead be:

`if (call->IsUnmanaged() && call->IsSuppressGCTransition())`

In other words, we should never try to omit the GCPOLL call.